src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h

```diff
@@ -35,7 +35,7 @@ class cRedstoneWireHandler : public cRedstoneHandler
 	const cVector3iArray GetTerracingConnectionOffsets(const Vector3i & a_Position)
 	{
 		cVector3iArray RelativePositions;
-		bool IsYPTerracingBlocked = cBlockInfo::IsSolid(m_World.GetBlock(a_Position + OffsetYP()));
+		bool IsYPTerracingBlocked = cBlockInfo::IsSolid(m_World.GetBlock(a_Position + OffsetYP())) && !cBlockInfo::IsTransparent(m_World.GetBlock(a_Position + OffsetYP()));
```

Comment:
Avoid calling `cWorld:GetBlock()` multiple times for the same block - store the returned value in a temporary variable.

(Each `GetBlock()` call needs to lock a mutex and search for the appropriate chunk, which is not exactly cheap)